src/dsl.jl

```diff
@@ -110,158 +50,207 @@ end
+`reaction_network` DSL, in particular, is described in more detail [here](@ref dsl_description).
+
+The `@reaction_network` statement is followed by a `begin ... end` block. Each line within the
+block corresponds to a single reaction. Each reaction consists of:
+- A rate (at which the reaction occur).
+- Any number of substrates (which are consumed by the reaction).
+- Any number of products (which are produced by the reaction).
 
-See [The Reaction DSL](@ref dsl_description) documentation for details on
-parameters to the macro.
 
 Examples:
+Here we create a basic SIR model. It contains two reactions (infections and recovery):
 ```julia
-# a basic SIR model, with name SIR
-sir_model = @reaction_network SIR begin
-    c1, s + i --> 2i
-    c2, i --> r
-end
-
-# a basic SIR model, with random generated name
 sir_model = @reaction_network begin
-    c1, s + i --> 2i
-    c2, i --> r
+    c1, S + I --> 2I
+    c2, I --> R
 end
+```
 
-# an empty network with name empty
-emptyrn = @reaction_network empty
-
-# an empty network with random generated name
-emptyrn = @reaction_network
+Next we create a self-activation loop. Here, a single component (`X`) activates its own production
+with a Michaelis-Menten function:
+```julia
+sa_loop = @reaction_network begin
+    mm(X,v,K), 0 --> X
+    d, X --> 0
+end
 ```
+This model also contain production and degradation reactions, where `0` denotes that there are 
+either no substrates or not products in a reaction.
+
+Options:
+The `@reaction_network` also accepts various options. These are inputs to the model that are not
+reactions. To denote that a line contain an option (and not a reaction), the line starts with `@`
+followed by the options name. E.g. an observable is declared using the `@observables` option.
+Here we create a polymerisation model (where the parameter `n` denotes the number of monomers in
+the polymer). We use the observable `Xtot` to track the total amount of `X` in the system. We also
+bundle the forward and backwards binding reactions into a single line.
+```julia
+polymerisation = @reaction_network begin
+    @observables Xtot ~ X + n*Xn
+    (kB,kD), n*X <--> Xn
+end
+```
+
+Notes:
+- `ReactionSystem`s creates through `@reaction_network` are considered complete (non-complete 
+systems can be created through the alternative `@network_component` macro).
+- `ReactionSystem`s creates through `@reaction_network`, by default, have a random name. Specific
+names can be designated as a first argument (before `begin`, e.g. `rn = @reaction_network name begin ...`).
+- For more information, please again consider Catalyst's documentation.
 
-ReactionSystems generated through `@reaction_network` are complete.
 """
-macro reaction_network(name::Symbol, ex::Expr)
-    :(complete($(make_reaction_system(
-        MacroTools.striplines(ex); name = :($(QuoteNode(name)))))))
+macro reaction_network(name::Symbol, network_expr::Expr)
```

Comment:
Basically, there are 4 different types of input (nothing, name, reactions, name + reactions). Each case is handled by a set of functions, which are now called from both `@eaction_network` and `@network_component`.